### DISCUSSION TITLE
Hive Alerter - Title Field - Dynamic Fields

### DISCUSSION THREAD

**Discussion by portalthiago** [jertel/elastalert2#676] [Q&A] [answered]
Hi @jertel and @nsano-rururu thx for all modifications on this project!
Im trying to use title with dynamic field like this using on "title" but with no luck.

The error return : "ERROR:elastalert:Error while running alert hivealerter: Error posting to TheHive: 400 Client Error: Bad Request for url: https://hive-hml.domain::443/api/alert"

Example:

  follow: True
  severity: 2
  status: 'New'
  source: 'elastalert'
  description: 'Sample description'
  tags: [ 'host.name','user.name' ] <<<== Works
  title: [ host.name ]    <<=== Please helpe me understand. 
  tlp: 3
  type: 'crowdstrike' <<<== Only Strings Saddly =/
  caseTemplate: "Crowdstrike" <<<== Works
if i just take it of any character for mark as atribut like ' " { [ to try user dynamic field just dont work. Can u help me?

Thx in Advance.
Best Regards.

![image](https://user-images.githubusercontent.com/20935653/151405297-257a4cf9-feb0-469f-9758-b46378a1a963.png)

**Comment by jertel** [comment, marked as answer]
Hello,

As mentioned [here](https://github.com/jertel/elastalert2/discussions/465#discussioncomment-1514013) TheHive alerter has limited support for dynamic value substitution into only a few fields. Unfortunately, the `type` field is one that is not dynamic. To support this you will need to issue a PR, following the [contribution guidelines](https://github.com/jertel/elastalert2/blob/master/CONTRIBUTING.md), or use an [Enhancement](https://elastalert2.readthedocs.io/en/latest/recipes/adding_enhancements.html). 

I do not know why this functionality was removed, but it appears to have occurred in between these two commits:

May 21, 2021: [c7bd38764c65175817b3b2b55ccd3175b26acb00](https://github.com/jertel/elastalert2/commit/c7bd38764c65175817b3b2b55ccd3175b26acb00)
and
May 22, 2022: [b8f3ea7009826d888e40a96f19aeb14d187de216](https://github.com/jertel/elastalert2/commit/b8f3ea7009826d888e40a96f19aeb14d187de216)

Regardless, the functionality would need to be re-added, re-tested, and documented in order to be accepted back into the project.

As far as the title goes, that is explained in the [TheHive alerter documentation](https://elastalert2.readthedocs.io/en/latest/ruletypes.html#thehive):

Specifically: 
![image](https://user-images.githubusercontent.com/16639611/151454639-56610a53-5c6a-4c8e-bcdf-ae6119dc73fb.png)

In other words, to customize TheHive alert title, you must use the `alert_subject` syntax as documentation [here](https://elastalert2.readthedocs.io/en/latest/ruletypes.html#alert-subject). If you do not add that syntax to your rule then it will default to the name of the rule for the title field, unless you specify some custom string for the title field directly.

**Reply by portalthiago** [reply]
Hey @jertel thx for the response, now i to send match information on title. When i comment the #title on the yml rule. it works. Thx for the help! I will ask my team about the possibility to help on this project! We use elastalert as much as we can! 

![image](https://user-images.githubusercontent.com/20935653/151592580-857fbbe1-4a28-4dda-8fc2-0f29522f1334.png)

**Reply by nsano-rururu** [reply]
@jertel 

>I do not know why this functionality was removed, but it appears to have occurred in between these two commits:

I saw the change, but it looks like I'm intentionally changing it.

Refactor TheHive alerter
https://github.com/jertel/elastalert2/commit/57de2a589d2acf35f4ffd3cc54261779e472bab8#diff-93c4ac73b454d85d4631ec52accb336d3cc319ada04dfa64ad14bc7242294c09

**Reply by portalthiago** [reply]
If this funcionallity for "descritpion" field could be used again, will use-it to create storys for the alerta as i already do on elastalert1.

Ex; 
A malware {{xpto}} was detected on host; {{host.name}} from domain: {{user.domain}} with the process {{process.name}} and was {{event,action}} with {{event.outcome}}. 

thx for all the help!

**Reply by portalthiago** [reply]
Hi, to finish the question, and help others, i finally understand. When i comment #title and #description i be able to send all info i need again. 
![image](https://user-images.githubusercontent.com/20935653/151605038-d0940138-51eb-47c7-899e-d58cd98d5e33.png)
